src/test/java/io/lettuce/core/cluster/models/partitions/RedisClusterNodeUnitTests.java

```diff
@@ -67,6 +69,42 @@ void shouldCopyNodeWithNonEmptySlots() {
         assertThat(copiedNode.getSlots()).containsExactly(1, 2);
     }
 
+    @Test
+    public void testHasSameSlotsAs() {
+
+        BitSet slots1 = new BitSet(SlotHash.SLOT_COUNT);
+        slots1.set(1);
+        slots1.set(2);
+
+        BitSet slots2 = new BitSet(SlotHash.SLOT_COUNT);
+        slots2.set(1);
+        slots2.set(2);
+
+        RedisClusterNode node1 = new RedisClusterNode(RedisURI.create("localhost", 6379), "nodeId1", true, "slaveOf", 0L, 0L,
+                0L, slots1, new HashSet<>());
+        RedisClusterNode node2 = new RedisClusterNode(RedisURI.create("localhost", 6379), "nodeId2", true, "slaveOf", 0L, 0L,
+                0L, slots2, new HashSet<>());
+
+        assertThat(node1.hasSameSlotsAs(node2)).isTrue();
+    }
+
+    @Test
+    public void testHasDifferentSlotsAs() {
```

Comment:
This test is OK, but it tests something different than what is explained in #2341 

In #2341 the problems is that one part of the logic uses `RedisClusterNode(RedisURI uri, String nodeId, boolean connected, String slaveOf, long pingSentTimestamp, long pongReceivedTimestamp, long configEpoch, BitSet slots, Set<NodeFlag> flags)` to create a RedisClusterNode, while another uses the `public RedisClusterNode(RedisClusterNode redisClusterNode)` constructor. Both are provided an empty BitSet.

When later on both resulting objects are compared with `.hasSameSlotsAs()` it would return `false`, when it should return `true`, because the first object would have an empty `slots` BitSet, while the second would have a null `slots` BitSet.

I think the proper test scenario is :

```java
public void testHasDifferentSlotsAs() {
        BitSet emptySlots = new BitSet(SlotHash.SLOT_COUNT);

        RedisClusterNode node1 = new RedisClusterNode(RedisURI.create("localhost", 6379), "nodeId1", true, "slaveOf", 0L, 0L,
                0L, emptySlots, new HashSet<>());
        RedisClusterNode node2 = new RedisClusterNode(node1);

        assertThat(node1.hasSameSlotsAs(node2)).isTrue();
```

You can test if this case fails when you remove your changes to `RedisClusterNode.java`

Reply:
I missed the part where we need to compare the original cluster node with the cloned object. Thanks for the feedback!

Reply:
Awesome!